src/main/java/me/minidigger/hangar/controller/VersionsController.java

```diff
@@ -135,19 +175,68 @@ public Object downloadRecommendedJar(@PathVariable Object author, @PathVariable
     }
 
     @Secured("ROLE_USER")
-    @RequestMapping("/{author}/{slug}/versions/{version}")
-    public Object publish(@PathVariable Object author, @PathVariable Object slug, @PathVariable Object version) {
-        return null; // TODO implement publish request controller
+    @PostMapping(value = "/{author}/{slug}/versions/{version:.+}", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public ModelAndView publish(@PathVariable String author,
```

Comment:
The whole method is kinda big that should be refactored. That should be done by having helper methods that doing the stuff below.